src/steamship_langchain/llms/openai.py

```diff
@@ -1,64 +1,107 @@
 import hashlib
+import json
 import logging
-from typing import Any, List, Mapping, Optional
+from collections import defaultdict
+from typing import Any, Dict, Generator, List, Mapping, Optional
 
-import tiktoken
-from langchain.llms.base import BaseLLM, Generation, LLMResult
-from pydantic import Field
+from langchain.llms.base import Generation, LLMResult
+from langchain.llms.openai import BaseOpenAI
+from pydantic import root_validator
 from steamship import Block, File, Steamship, SteamshipError
 from steamship.data import TagKind, TagValueKey
 
-
-class OpenAI(BaseLLM):
+PLUGIN_HANDLE: str = "gpt-3"
+ARGUMENT_WHITELIST = {
+    "client",
+    "model_name",
+    "temperature",
+    "max_tokens",
+    "top_p",
+    "frequency_penalty",
+    "presence_penalty",
+    "n",
+    "best_of",
+    "model_kwargs",
+    "openai_api_key",
+    "batch_size",
+    "request_timeout",
+    # "logit_bias",
+    "max_retries",
+    "callback_manager",
+    "cache",
+    "verbose",
+}
+
+
+class OpenAI(BaseOpenAI):
     """Implements LangChain LLM interface in a Steamship-compatible fashion, allowing use in chains/agents as required.
 
     NOTE: This provides a **synchronous** interaction with the LLM backend.
     """
 
-    client: Steamship
-    plugin_handle: str = Field(default="gpt-3", const=True)
-    batch_size: int = Field(default=20)
-    temperature: float = 0.8
-    max_words: int = 256
-    n: int = 1
-    best_of: int = 1
+    client: Steamship  # We can use validate_environment to add the client here
     batch_task_timeout_seconds: int = 10 * 60  # 10 minute limit on generation tasks
 
-    @property
-    def _llm_type(self) -> str:
```

Comment:
curious: why get rid of `_llm_type()` ?

Reply:
To me, llm_type is still OpenAI right? It's not like we're a new llm type that is only offered through steamship.